server/src/main/java/org/elasticsearch/transport/TcpChannel.java

```diff
@@ -79,4 +86,40 @@
      * @param listener to be executed
      */
     void addConnectListener(ActionListener<Void> listener);
+
+    /**
+     * Returns stats about this channel
+     */
+    ChannelStats getChannelStats();
+
+    class ChannelStats {
+
+        private volatile long lastReadTime;
+        private volatile long lastWriteTime;
+
+        public ChannelStats() {
```

Comment:
I wonder if it makes sense to have a `LongSupplier` passed to it then you don't need to pass the value to the mark methods?

Reply:
This feels similar to the issue with described in (https://github.com/elastic/elasticsearch/pull/35441#discussion_r233150475). If `ChannelStats` must have a `LongSupplier`, then every different implementation must have `ThreadPool` passed around everywhere. Which I would kind of like to avoid doing.

Reply:
well it must have some kind of `LongSupplier` that's the point. The threadpool can be one impl of it.